pymonad/either.py

```diff
@@ -49,9 +49,9 @@ def amap(self: 'Either[M, Callable[[S], T]]', monad_value: 'Either[M, S]') -> 'E
         else:
             return self.__class__(self.value(monad_value.value), (None, True))
 
-    def bind(
-            self: 'Either[M, S]', kleisli_function: Callable[[S], 'Either[M, T]']
-    ) -> 'Either[M, T]':
+    def bind[Err, Val](
```

Comment:
Unless you defined Err and Val in a previous PR, this will likely break the typing. S and T are referenced via TypeVar. Err and Val are not.
Also, you need to consider backward compatibility, this way of defining generics isn't supported earlier than python 3.13